mobxjs/mobx-state-tree
# Feat/add formatting config

## What does this PR do and why?

Closes https://github.com/mobxjs/mobx-state-tree/issues/2219.

This changes our Prettier config to match the [config for MobX](https://github.com/mobxjs/mobx/blob/main/.prettierrc).

It also runs `bun prettier:write` against the whole codebase, so there's a huge diff here.

We actually have [Husky](https://github.com/typicode/husky) set up to do this as a pre-commit hook already. I think the reason we've had a bunch of diffs is because some of the source files hadn't been given the correct formatting at some point, so as we've made changes throughout the codebase, the Husky hook has formatted them for us.

I've also added a step to run `prettier:check` in CI, so if people skip the hook, we'll see the problem in PRs.

## Steps to validate locally

You should have `[typescript]editor.formatOnSave: true` in `.vscode/settings.json`. Turn that to `false`, then make a change to a file that breaks the Prettier config.

Save the file, add it, and commit it. Husky should clean up your code. 

`bun test` should pass, and `bun run build` should pass and work.

CI should pass on this branch, including the new `check-prettier` job.
